File: FeedItForward/backend/app/schemas/consumer.py
from pydantic import BaseModel
from typing import Optional

from .user import User, UserCreate, UserUpdate, Role

class Consumer(BaseModel):
    consumer_id: int
    priority: bool = False

    user_id: int
    user: User

    class Config:
        orm_mode = True

class ConsumerCreate(UserCreate):
    role: Role = Role.CONSUMER

class ConsumerUpdate(UserUpdate):
    consumer_id: int
    priority: Optional[bool]

File: FeedItForward/frontend/src/schemas/consumer.ts
import { UserCreate, UserDisplay, UserUpdate } from "./user";

export interface Consumer {
  consumer_id: number;
  priority: boolean;

  user_id: number;
  user: UserDisplay;
}

export interface ConsumerCreate extends UserCreate {}

export interface ConsumerUpdate extends UserUpdate {
  consumer_id: number;
  priority?: boolean;
}
